### PR TITLE
LCAM-1045

### PR DESCRIPTION
**Pull request by Mhowell494**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-1045)

- Fixed SonarCloud warnings
- Improved project structure and identifier names.
- Refactored the ProceedingsMapper by extracting a new method.
- New controller and service to support workflow for updates to the crown court tab in MAAT
- New dependency added for annotation processing to prevent warnings on custom properties.
- New and improved test scenarios to support the above changes.

### CODE REVIEW

File: maat-orchestration/build.gradle [hunk 1]
```diff
@@ -37,6 +37,8 @@ dependencies {
 	implementation "uk.gov.justice.service.laa-crime:crime-commons-spring-boot-starter-rest-client:$versions.crimeCommonsVersion"
 	implementation "uk.gov.justice.service.laa-crime:crime-commons-classes:$versions.crimeCommonsVersion"
 
+	annotationProcessor "org.springframework.boot:spring-boot-configuration-processor"
```
Comment (skasthuri36326) [inline maat-orchestration/build.gradle:40]: Spring boot configuration processor is inherited from the crime commons rest client lib. If we use the fat jar then we can remove line 40.

Change Line 37 to :       ` implementation "uk.gov.justice.service.laa-crime:crime-commons-spring-boot-starter-rest-client:$versions.crimeCommonsVersion:all"`

and we can remove Line 40

Comment (Mhowell494) [reply maat-orchestration/build.gradle]: Good shout :)

File: maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/controller/CrownCourtController.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    @ApiResponse(responseCode = "200",
+            content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE,
+                    schema = @Schema(implementation = ApiUpdateApplicationRequest.class)
+            )
+    )
+    @DefaultHTTPErrorResponse
+    public ResponseEntity<ApplicationDTO> updateCrownCourt(
+            @Valid @RequestBody WorkflowRequest workflowRequest,
+            @Parameter(description = "Used for tracing calls") @RequestHeader(value = LAA_TRANSACTION_ID, required = false) String laaTransactionId) {
+        log.info("Received request to update crown court");
```
Comment (skasthuri36326) [inline maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/controller/CrownCourtController.java:43]: We agreed on logging the laaTransactionId at all the Orchestration entrypoint for tracing.

File: maat-orchestration/src/main/resources/schemas/crown-court/apiUpdateCrownCourtResponse.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "id": "apiUpdateOutcomeResponse.json",
```
Comment (venkat980) [inline maat-orchestration/src/main/resources/schemas/crown-court/apiUpdateCrownCourtResponse.json:3]: Can we keep the id and the file name consistent please?

**Review by venkat980 — CHANGES_REQUESTED** [review]
Overall excellent job! Just couple of trivial comments. Thanks!

**Review by skasthuri36326 — APPROVED** [review]

**Pull request merged**